vnet/lib/vnet/models/network.rb

```diff
@@ -35,6 +42,13 @@ class Network < Base
     # 0011_assoc_interface
     interface_networks: :destroy
 
+    def before_destroy
+      # the association_dependencies plugin doesn't allow us to destroy because it's a many to many relation
+      self.ip_leases.each { |lease| lease.destroy }
```

Comment:
I don't think that is possible. interface and mac_lease delete ip_lease through assoc_dep. That means that at that time ip_lease needs to clean up it's ip_address through assoc_dep.

If we give ip_address an assoc_dep to destroy ip_lease, then we introduce a circular dependency. ip_address will call before_destroy which destroys ip_lease which calls before_destroy which destroys ip_address.

Just to make sure, I tried it and this happens:

```
  2) Vnet::Models::Network destroy deletes associated ip leases and ip addresses
     Failure/Error: nw.destroy

     Sequel::NoExistingObject:
       Attempt to update object did not result in a single row modification (SQL: UPDATE `ip_addresses` SET `network_id` = 3, `ipv4_address` = 3232235781, `created_at` = '2017-04-25 06:00:52', `updated_at` = '2017-04-25 06:00:52', `deleted
_at` = '2017-04-25 06:00:52', `is_deleted` = 9 WHERE ((`ip_addresses`.`deleted_at` IS NULL) AND (`id` = 9)) LIMIT 1)
```